js/parts/Chart.js

```diff
@@ -446,16 +446,16 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
                     if (serie.updateTotals) {
                         serie.updateTotals();
                     }
-                    redrawLegend = true;
                 }
+                redrawLegend = true; // #2165
```

Comment:
Additional tests to redraw legend only when a formatter is defined.

Reply:
Done